addon/-private/system/model/model.js

```diff
@@ -587,8 +587,9 @@ const Model = Ember.Object.extend(Ember.Evented, {
   },
 
   /**
+    Unloads the record from the store. Only non-dirty records can be unloaded.
```

Comment:
I'd remove the second sentence and add "This will cause the record to be destroyed and freed up for garbage collection.".